src/Soil/Biogeochemistry/Biogeochemistry.jl

```diff
@@ -137,6 +158,311 @@ function DETECTModelParameters(;
     )
 end
 
-include("./co2_parameterizations.jl")
+
+
+
+#################################
+
+"""
+    DETECTModel
+
+A model for simulating the production and transport of CO₂ in the soil with dynamic
+source and diffusion terms.
+"""
+struct DETECTModel{FT, PS, D, BC, S, DT} <: ClimaLSM.AbstractModel{FT}
+    "the parameter set"
+    parameters::PS 
+    "the soil domain, using ClimaCore.Domains"
+    domain::D
+    "the boundary conditions, of type AbstractBoundaryConditions"
+    boundary_conditions::BC 
+    "A tuple of sources, each of type AbstractSource"
+    sources::S
+    " Drivers"
+    driver::DT
+end
+
+
+"""
+DETECTModel{FT}(;
+        parameters::DETECTParameters{FT},
+        domain::ClimaLSM.AbstractDomain,
+        boundary_conditions::NamedTuple,
+        sources::Tuple,
+    ) where {FT}
+
+A constructor for `DETECTModel`.
+"""
+function DETECTModel{FT}(;
+    parameters::DETECTParameters{FT},
+    domain::ClimaLSM.AbstractDomain,
+    boundary_conditions::BC,
+    sources::Tuple,
+    drivers::DT
+) where {FT,BC, DT}
+    args = (parameters, domain, boundary_conditions, sources, drivers)
+    DETECTModel{FT, typeof.(args)...}(args...)
+end
+
+ClimaLSM.name(model::DETECTModel) = :DETECT;
```

Comment:
if our model names are grouped like `soil`, `snow`, etc, maybe this name could be `soilco2`? not sure. just wondering if DETECT is the best descriptor.

Reply:
soilco2 sounds good!

Reply:
I replaced all `DETECT` occurrences with `soilco2` for now, we can always change later.